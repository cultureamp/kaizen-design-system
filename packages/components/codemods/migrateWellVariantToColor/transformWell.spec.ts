// import fs from "fs"
// import path from "path"
// import ts from "typescript"
// import { updateFileContents } from "./transformWell"

// // This function is used to parse a stringified JSX element into an AST
// function parseJsx(jsx: string): ts.SourceFile {
//   return ts.createSourceFile(
//     "tempFile.tsx",
//     jsx,
//     ts.ScriptTarget.Latest,
//     true,
//     ts.ScriptKind.TSX
//   )
// }

// // describe("updateFileContents", () => {
// //   it("should update Well components imported from @kaizen/components", () => {
// //     const filePath = path.resolve(
// //       path.join(__dirname, "./__fixtures__/Well.tsx")
// //     )
// //     const fileContent = fs.readFileSync(filePath, "utf8")
// //     const sourceFile = parseJsx(fileContent)

// //     expect(updateFileContents(sourceFile, "Well")).toMatchSnapshot()
// //   })
// //   it("should update Well components imported from @kaizen/component/path", () => {
// //     const filePath = path.resolve(
// //       path.join(__dirname, "./__fixtures__/WellV3.tsx")
// //     )
// //     const fileContent = fs.readFileSync(filePath, "utf8")
// //     const sourceFile = parseJsx(fileContent)

// //     expect(updateFileContents(sourceFile, "Well")).toMatchSnapshot()
// //   })
// //   it("should update aliased Well components imported from @kaizen/component", () => {
// //     const filePath = path.resolve(
// //       path.join(__dirname, "./__fixtures__/WellAlias.tsx")
// //     )
// //     const fileContent = fs.readFileSync(filePath, "utf8")
// //     const sourceFile = parseJsx(fileContent)

// //     expect(updateFileContents(sourceFile, "KaizenWell")).toMatchSnapshot()
// //   })
// // })
