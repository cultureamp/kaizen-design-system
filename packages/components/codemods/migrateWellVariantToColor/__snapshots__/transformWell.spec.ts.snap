// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateFileContents should not update Well components imported from @kaizen/draft-well or legacy packages 1`] = `
"import * as React from "react";
// @ts-ignore
import { Well } from "@kaizen/draft-well";
export const LegacyWells = (): JSX.Element => (<>
    <Well variant="default">Test 1 should have default variant</Well>
    <Well>Test 2 should have no additional props</Well>
    <Well variant="informative">Test 3 should have informative variant</Well>
    <Well id="test-id" variant="informative">
      Test 4 should have informative variant and an id prop
    </Well>
  </>);
"
`;

exports[`updateFileContents should update Well components imported from @kaizen/component/path 1`] = `
"import * as React from "react";
// @ts-ignore
import { Well } from "@kaizen/components/v3";
/* eslint-disable react/jsx-curly-brace-presence */
export const TestWellComponent = (): JSX.Element => {
    const informativeVar = "informative";
    return (<>
      <Well color="gray">Test 1 should have color gray</Well>
      <Well color="gray">Test 2 should have color gray</Well>
      <Well color="blue">Test 3 should have color blue</Well>
      <Well id="test-id" color="blue">
        Test 4 should have color blue and an id prop
      </Well>
      <Well color="blue">Test 5 should have color blue</Well>
      <Well variant={informativeVar}>
        Test 6 should have variant informativeVar
      </Well>
      <Well color="blue">Test 7 should have color blue</Well>
    </>);
};
/* eslint-enable react/jsx-curly-brace-presence */
"
`;

exports[`updateFileContents should update Well components imported from @kaizen/components 1`] = `
"import * as React from "react";
// @ts-ignore
import { Well } from "@kaizen/components";
/* eslint-disable react/jsx-curly-brace-presence */
export const TestWellComponent = (): JSX.Element => {
    const informativeVar = "informative";
    return (<>
      <Well color="gray">Test 1 should have color gray</Well>
      <Well color="gray">Test 2 should have color gray</Well>
      <Well color="blue">Test 3 should have color blue</Well>
      <Well id="test-id" color="blue">
        Test 4 should have color blue and an id prop
      </Well>
      <Well color="blue">Test 5 should have color blue</Well>
      <Well variant={informativeVar}>
        Test 6 should have variant informativeVar
      </Well>
      <Well color="blue">Test 7 should have color blue</Well>
    </>);
};
/* eslint-enable react/jsx-curly-brace-presence */
"
`;

exports[`updateFileContents should update aliased Well components imported from @kaizen/component 1`] = `
"import * as React from "react";
// @ts-ignore
import { Well as KaizenWell } from "@kaizen/components";
/* eslint-disable react/jsx-curly-brace-presence */
export const TestWellComponent = (): JSX.Element => {
    const informativeVar = "informative";
    return (<>
      <KaizenWell color="gray">Test 1 should have color gray</KaizenWell>
      <KaizenWell color="gray">Test 2 should have color gray</KaizenWell>
      <KaizenWell color="blue">
        Test 3 should have color blue
      </KaizenWell>
      <KaizenWell id="test-id" color="blue">
        Test 4 should have color blue and an id prop
      </KaizenWell>
      <KaizenWell color="blue">
        Test 5 should have color blue
      </KaizenWell>
      <KaizenWell variant={informativeVar}>
        Test 6 should have variant informativeVar
      </KaizenWell>
      <KaizenWell color="blue">Test 7 should have color blue</KaizenWell>
    </>);
};
/* eslint-enable react/jsx-curly-brace-presence */
"
`;
